cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -142,6 +175,57 @@ func (c *Controller) create(pendingPoolCount int, spc *apis.StoragePoolClaim) er
 	return nil
 }
 
+// cleanUPSPCChild will remove pool related resources and finalizers on BDC
+func (c *Controller) cleanUpSPCChild(spc *apis.StoragePoolClaim) error {
+	// Delete csp related to spc
+	err := DeleteCSPList(spc.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete csp belongs to spc %s", spc.Name)
+	}
+	namespace := env.Get(env.OpenEBSNamespace)
+	err = blockdeviceclaim.RemoveFinalizersOnBDCList(
+		namespace,
+		string(apis.StoragePoolClaimCPK)+"="+spc.Name,
+		spcv1alpha1.SPCFinalizer,
+	)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"failed to remove finalizer on blockdeviceclaims related spc %s",
+			spc.Name,
+		)
+	}
+	_, err = spcv1alpha1.AddOrRemoveSPCFinalizer(
+		spc,
+		spcv1alpha1.SPCFinalizer,
+		spcv1alpha1.RemoveSPCFinalizer,
+	)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer on spc %s", spc.Name)
+	}
+	return nil
+}
+
+// TODO: Need to move to corresponding cstor pool package
+
+// DeleteCSPList deletes the list of csp related to spc
+func DeleteCSPList(spcName string) error {
+	cspClient := csp.NewKubeClient()
+	err := cspClient.DeleteCollection(
+		metav1.ListOptions{
+			LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spcName,
+		},
+		&metav1.DeleteOptions{},
+	)
+	if k8serror.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete collection csps related to spc %s", spcName)
+	}
+	return errors.Errorf("Deleted cstorpools related to spc %s", spcName)
```

Comment:
Why are we returning an error here?

Reply:
So we just triggered delete call and not verifying whether really csp's are deleted or not in these cases we are returning error